File: krypto/02/knapsack-crypto/src/main/java/algorithm/SimpleKeyGenerator.java
package algorithm;

import lombok.*;

@Data
@AllArgsConstructor
public class SimpleKeyGenerator {

    private int keySize = 8;
    private long multiplier, modulus;

    public SimpleKeyGenerator() {
        this(4571L, 7187L);
    }

    public SimpleKeyGenerator(long multiplier, long modulus) {
        this.multiplier = multiplier;
        this.modulus = modulus;
    }

    long[] generateDefaultPrivateKey() {
        long[] newPrivateKey = new long[keySize];
        long keyElement = 1;
        for (int i = 0; i < newPrivateKey.length; i++) {
            keyElement *= 2;
            newPrivateKey[i] = keyElement;
        }
        return newPrivateKey;
    }

    long[] generatePublicKey(long[] privateKey) {
        long[] newPublicKey = new long[keySize];
        for (int i = 0; i < privateKey.length; i++) {
            newPublicKey[i] = (privateKey[i] * multiplier) % modulus;
        }
        return newPublicKey;
    }
}

File: krypto/zad2py/simple_key_generator.py
class SimpleKeyGenerator:
    def __init__(self):
        self.multiplier = 4571
        self.modulus = 18446744073709551615
        self.key_size = 8

    def generate_default_private_key(self):
        new_private_key = []
        key_element = 18446744073709551615
        for i in range(self.key_size):
            key_element /= 2
            new_private_key.append(int(key_element))
        return list(reversed(new_private_key))

    def generate_default_public_key(self, private_key):
        new_public_key = []
        index = 0
        for i in private_key:
            new_public_key.append((i * self.multiplier) % self.modulus)
            index += 1
        return new_public_key
